### DISCUSSION TITLE
Getting cluster metadata

### DISCUSSION THREAD

**Discussion by stefanofornari** [ing-bank/cassandra-jdbc-wrapper#70] [Ideas]
I have a use case where I need to know the toke range each node of the cluster manages. This is done with the java driver as follows:

```
String keyspace = "mykeyspace";
for(Host host : cluster.getMetadata().getAllHosts()) {
    Set<TokenRange> hostRanges = cluster.getMetadata().getTokenRanges(keyspace, host);
}
```
I could not find a way to get the same information with the java wrapper. This forces me to use both, the java driver to get the ranges and then the wrapper for everything else. Would it be an idea to have a `getClusterMetadata()` returning the same as `cluster.getMetadata()` above? of course it would be specific to cassandra, but I do not think this would be a problem.

**Comment by maximevw** [comment]
Hello @stefanofornari 

Normally, the cluster metadata should already be accessible via the `CassandraConnection` using the method [`getClusterMetadata()`](https://javadoc.io/static/com.ing.data/cassandra-jdbc-wrapper/4.12.0/com/ing/data/cassandra/jdbc/CassandraConnection.html#getClusterMetadata--).

Please note that the code you provided seems to use an old version of the Java driver, so the way to retrieve the token ranges is now slightly different in the version 4.x. Using the method specified above, you should get something like this:
```java
CassandraConnection connection; // Assuming you already declared your connection previously.
for (Node node : connection.getClusterMetadata().getNodes().values()) {
        Set<TokenRange> tokenRanges = connection.getClusterMetadata().getTokenMap().get().getTokenRanges(node);
}
```
